storage_scrubber/src/pageserver_physical_gc.rs

```diff
@@ -608,6 +624,27 @@ async fn gc_timeline(
     Ok(summary)
 }
 
+fn validate_index_part_with_offloaded(
+    ttid: TenantShardTimelineId,
+    index_part: &IndexPart,
+    offloaded: &OffloadedTimelineManifest,
+) {
+    if let Some(archived_at_index_part) = index_part.archived_at {
+        if archived_at_index_part
+            .signed_duration_since(offloaded.archived_at)
+            .num_seconds()
+            != 0
+        {
+            tracing::warn!(%ttid, "index-part archived_at={} differs from manifest archived_at={}", archived_at_index_part, offloaded.archived_at);
+        }
+    } else {
+        tracing::warn!(%ttid, "Timeline offloaded in manifest but not archived in index-part");
+    }
```

Comment:
What about unarchiving though?

```suggestion
    } else {
        // Archiving always modifies index part before manifest.
        tracing::warn!(%ttid, "Timeline offloaded in manifest but not archived in index-part");
    }
```